### ISSUE TITLE
[Bug] Precompile speed is very slow

### ISSUE THREAD

**Issue by jimoosciuc** [sgl-project/sglang-jax#117]
### Checklist

- [ ] 1. I have searched related issues but cannot get the expected help.
- [ ] 2. The bug has not been fixed in the latest version.
- [ ] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.
- [ ] 4. If the issue you raised is not a bug but a question, please raise a discussion at https://github.com/sgl-project/sgl-jax/discussions/new/choose Otherwise, it will be closed.
- [ ] 5. Please use English, otherwise it will be closed.

### Describe the bug

Even with persistent caching, precompile speed is still very slow!

### Launch Server
```
JAX_COMPILATION_CACHE_DIR=/tmp/jit_cache  python3 -u -m sgl_jax.launch_server --model-path Qwen/Qwen3-8B --trust-remote-code  --dist-init-addr=0.0.0.0:10011 --nnodes=1  --tp-size=4 --device=tpu --random-seed=3 --node-rank=0 --mem-fraction-static=0.8 --chunked-prefill-size=8192 --download-dir=/tmp --dtype=bfloat16 --jax-precompile-decode-bs-paddings 1 2 4 8 16 32 64 128 --max-running-requests 256 --max-total-tokens 257536 --skip-server-warmup --attention-backend=fa --jax-precompile-prefill-token-paddings 128 256 512 8192 --page-size=64
```
### Result
[2025-09-03 04:38:57] [EXTEND] begin to precompile
[2025-09-03 04:38:57] [EXTEND] precompile (bs=128, num_tokens=128)
[2025-09-03 04:39:41] [EXTEND] precompile (bs=128, num_tokens=256)
[2025-09-03 04:40:24] [EXTEND] precompile (bs=128, num_tokens=512)
[2025-09-03 04:41:12] [EXTEND] precompile (bs=128, num_tokens=8192)
[2025-09-03 04:42:06] [EXTEND] precompile finished in 189 secs
[2025-09-03 04:42:06] [DECODE] begin to precompile
[2025-09-03 04:42:06] [DECODE] precompile (bs=1)
[2025-09-03 04:42:43] [DECODE] precompile (bs=2)
[2025-09-03 04:43:23] [DECODE] precompile (bs=4)
[2025-09-03 04:44:04] [DECODE] precompile (bs=8)
[2025-09-03 04:44:43] [DECODE] precompile (bs=16)
[2025-09-03 04:45:25] [DECODE] precompile (bs=32)
[2025-09-03 04:46:09] [DECODE] precompile (bs=64)
[2025-09-03 04:46:54] [DECODE] precompile (bs=128)
[2025-09-03 04:47:38] [DECODE] precompile finished in 331 secs
[2025-09-03 04:47:38] [Scheduler] completes worker precompile.

### Environment

NONE
